comps/integrations/haystack/src/opea_haystack/embedders/tei/text_embedder.py

```diff
@@ -0,0 +1,144 @@
+from typing import Any, Dict, List, Optional, Union
+
+from haystack import component, default_from_dict, default_to_dict
+
+from opea_haystack.utils import url_validation, OPEABackend
+
+from .truncate import EmbeddingTruncateMode
+
+_DEFAULT_API_URL = "http://localhost:6000/embed"
+
+
+@component
+class OPEATextEmbedder:
+    """
+    A component for embedding strings using embedding models provided by
+    [OPEA](https://opea.dev).
+
+    For models that differentiate between query and document inputs,
+    this component embeds the input string as a query.
+
+    Usage example:
+    ```python
+    from opea_haystack.embedders.tei import OPEATextEmbedder
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = OPEATextEmbedder(api_url="http://localhost:6000")
+    text_embedder.warm_up()
+
+    print(text_embedder.run({"text": text_to_embed})
```

Comment:
I prefer `"text"` but currently the component expects `"inputs"`. I suggest to change the `run` method's signature.